## Input data


In [1]:
durability_data = [
    1.5, 1.7, 1.6, 1.4, 1.5, 1.7, 1.6, 1.5, 1.4, 1.6,
    1.7, 1.5, 1.6, 1.4, 1.7, 1.5, 1.6, 1.4, 1.5, 1.7
]

## Calculate confidence interval (sample standard deviation)

#a 99% confidence interval for the mean durability using the sample standard deviation and the t-distribution.


In [ ]:
## The t-distribution is used when the population standard deviation is unknown and the sample size is small (or the data is assumed to be normally distributed). This is because the t-distribution accounts for the additional uncertainty introduced by estimating the population standard deviation from the sample data.

In [5]:
import numpy as np
from scipy import stats



# 1.sample mean
sample_mean = np.mean(durability_data)
print(f"\nSample mean: {sample_mean:.4f}")

# 2.sample standard deviation
sample_std = np.std(durability_data, ddof=1) # Use ddof=1 for sample standard deviation
print(f"Sample standard deviation: {sample_std:.4f}")

# 3.degrees of freedom
n = len(durability_data)
degrees_of_freedom = n - 1
print(f"Degrees of freedom: {degrees_of_freedom}")

# 4.critical t-score
confidence_level = 0.99
alpha = 1 - confidence_level
critical_t_score = stats.t.ppf(1 - alpha/2, degrees_of_freedom)
print(f"Critical t-score for 99% confidence: {critical_t_score:.4f}")

# 5.margin of error
margin_of_error = critical_t_score * (sample_std / np.sqrt(n))
print(f"Margin of error: {margin_of_error:.4f}")

# 6.99% confidence interval
confidence_interval_lower = sample_mean - margin_of_error
confidence_interval_upper = sample_mean + margin_of_error

# 8.calculated 99% confidence interval
print(f"\n99% Confidence Interval for the mean durability (using t-distribution): ({confidence_interval_lower:.4f}, {confidence_interval_upper:.4f})")


Sample mean: 1.5550
Sample standard deviation: 0.1099
Degrees of freedom: 19
Critical t-score for 99% confidence: 2.8609
Margin of error: 0.0703

99% Confidence Interval for the mean durability (using t-distribution): (1.4847, 1.6253)


## Calculate confidence interval (known population standard deviation)



In [ ]:
#Rationale for using the z-distribution: The z-distribution is used when the population standard deviation is known. In this case, the sample mean's distribution can be approximated by a normal distribution due to the Central Limit Theorem (especially for large sample sizes, though applicable here with known population std), and the z-score is used to determine critical values.

In [6]:
from scipy import stats
import numpy as np



# 1.Assume known population standard deviation
population_std = 0.2

# 2.z-score for a 99% confidence level
confidence_level = 0.99
alpha = 1 - confidence_level
# The z-score for a 99% confidence interval is the value z such that P(-z < Z < z) = 0.99
# This corresponds to the 0.995 percentile (1 - alpha/2)
critical_z_score = stats.norm.ppf(1 - alpha/2)
print(f"\nCritical z-score for 99% confidence: {critical_z_score:.4f}")

# 3.margin of error using the known population standard deviation
# We need the sample size 'n' and the sample mean 'sample_mean' from previous steps.
# Assuming 'n' and 'sample_mean' are available from the previous kernel state.
# If not, they would need to be recalculated from 'durability_data'.
# For this task, we will use the values from the previous execution context.
n = len(durability_data) # Assuming durability_data is available
sample_mean = np.mean(durability_data) # Assuming durability_data is available

margin_of_error_z = critical_z_score * (population_std / np.sqrt(n))
print(f"Margin of error (using z-distribution): {margin_of_error_z:.4f}")

# 4. 99% confidence interval using the sample mean and the new margin of error
confidence_interval_lower_z = sample_mean - margin_of_error_z
confidence_interval_upper_z = sample_mean + margin_of_error_z

print(f"\n99% Confidence Interval for the mean durability (using z-distribution): ({confidence_interval_lower_z:.4f}, {confidence_interval_upper_z:.4f})")


Critical z-score for 99% confidence: 2.5758
Margin of error (using z-distribution): 0.1152

99% Confidence Interval for the mean durability (using z-distribution): (1.4398, 1.6702)


## Summary:

### Data Analysis Key Findings

*   The sample mean durability of the print heads is 1.5550 million characters.
*   The sample standard deviation is 0.1099 million characters.
*   Using the t-distribution (when the population standard deviation is unknown), the 99% confidence interval for the mean durability is (1.4847, 1.6253) million characters. The margin of error for this interval is 0.0703 million characters.
*   Assuming a known population standard deviation of 0.2 million characters and using the z-distribution, the 99% confidence interval for the mean durability is (1.4398, 1.6702) million characters. The margin of error for this interval is 0.1152 million characters.


